enhancements/network/ovs-hardware-offload.md

```diff
@@ -0,0 +1,242 @@
+`ovs-vsctl set Open_vSwitch . other_config:tc-policy=skip_sw`. The default
+value is `none`, optional values can be one of `none`, `skip_sw` and `skip_hw`.
+This field is only relevant if `hw-offload` is enabled.
+
+#### Multus
+
+- Multus will support specifying net-attach-def with customized annotations for
+[default cluster network in pod annotations](https://github.com/intel/multus-cni/blob/master/doc/configuration.md#specify-default-cluster-network-in-pod-annotations).
+Customized annotations will contain a resource name requesting SR-IOV device.
+
+#### SR-IOV Operator
+
+- SR-IOV Operator will support new device type `switchdev` in Operator
+[SriovNetworkNodePolicy API](https://github.com/openshift/sriov-network-operator/blob/master/pkg/apis/sriovnetwork/v1/sriovnetworknodepolicy_types.go#L33).
+
+- SR-IOV config daemon (a DaemonSet managed by SR-IOV Operator) will support
+changing e-switch mode from `legacy` to `switchdev` on the PF device and
+exposing VF representor information in `SriovNetworkNodeState` status.
+
+- SR-IOV network device plugin (a Device Plugin DaemonSet managed by SR-IOV
+Operator) supports advertising VF resource to kubelet.
+
+#### Machine Config Operator (MCO) [optional]
+
+- MCO will be used to create linux bond interface on two identical PFs that
+are capable of doing OVS hardware offload and create vlan interfaces on bond
+interface. The supported bond modes are `Active-Backup`, `Active-Active`
+and `LACP`. This is not required if OVS Hardware Offload is used on PF
+device directly.
+
+### User Stories
+
+#### Story 1
+
+Workflow of creating a SR-IOV pod using OVS hardware offload:
+- [optional] Deploy a baremetal cluster with API network on bonded PFs
+- Enable OVS hardware offload for OpenvSwitch instances via CNO
+- Install SR-IOV Operator from Operator Hub
+- Provision PFs to desired number of VFs via SR-IOV Operator
+- Create a pod specifying cluster network with customized net-attach-def
+
+### Implementation Details/Notes/Constraints
+
+- When enabling OVS Hardware Offload with SR-IOV VF LAG, PFs should first be
+provisioned with desired number of VFs, then configure PFs to `switchdev`
+mode, linux bond interface shall be created after above two steps. Since linux
+bond configuration can be applied via ignition file, it's important that SR-IOV
+Operator (installed in day 2) makes sure `switchdev` configurations are applied
+before linux bond configuration, this probably requires a node rebooting to
+guarantee the order.
+
+- When enabling OVS `hw-offload` option for OpenvSwitch daemon, it is required
+that ovsdb is created first.
+
+### Risks and Mitigations
+
+RHEL-8.3 is not used as base image for RHCOS when 4.6 gets released. RHEL-8.3
+contains kernel and driver changes to fully support OVS Hardware Offload with
+connection track feature.
+
+### Test Plan
+
+- The changes in each relevant component must pass full e2e suite.
+- OVS hardware offload functional tests must pass on supported NICs.
```

Comment:
What are these "OVS hardware offload functional tests"?

So QE will have hardware to test this with? Do they have this already?

We don't plan to do full e2e runs on clusters with hardware offload configured?

Reply:
There is no defined functional tests in any of ovs hw offload related repos, the basic functional tests I was thinking includes following steps:
1) setup a baremetal cluster with ovs hw offload enabled.
2) create a pod with SR-IOV device working in switchdev mode
3) check the ovs flows for the pod are offloaded to hardware NIC and traffic between pods are not blocked.

QE has the same model of Mellanox CX-5 NIC as we had in baremetal CI, it will be used for running the ovs hw offload testing. The NIC is currently used for SR-IOV testing.

Regarding e2e with hardware offload enabled, it will be added as in current OVN CI once the relevant change is ready.